src/responder/secrets/local.c

```diff
@@ -332,6 +333,26 @@ static int local_db_check_containers(TALLOC_CTX *mem_ctx,
     return ret;
 }
 
+static int local_db_check_containers_nested_level(struct local_context *lctx,
+                                                  struct ldb_dn *leaf_dn)
+{
+    int nested_level;
+
+    /* We need do not care for the synthetic containers that constitute the
+     * base path (cn=<uidnumber>,cn=user,cn=secrets). */
+    nested_level = ldb_dn_get_comp_num(leaf_dn) - 3;
+    if (nested_level > lctx->max_containers_nested_level) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "Cannot create a nested container of depth %d as the maximum"
+              "allowed number of nested containers is %d.\n",
+              nested_level, lctx->max_containers_nested_level);
+
+        return EMEDIUMTYPE;
```

Comment:
Could we please add a new secrets-specific error code here so that we can have a nicer error message as well?